src/layers/AreaLayer.ts

```diff
@@ -0,0 +1,287 @@
+      __________________________         __________________________
+                                         
+                                       X                            X
+
+      Once we have these zoom extents SC in time t = 1, we can use our scales (which are also at t = 1)
+      and use scale.invert to get the zoom extents DC which correspond to the zoom extents SC at time
+      t = T.
+      */
+      const topLeftSC_0 = traceLayer.getNewPointInverse!(zoomExtentsSC.x[0], zoomExtentsSC.y[0], t);
+      const bottomRightSC_0 = traceLayer.getNewPointInverse!(zoomExtentsSC.x[1], zoomExtentsSC.y[1], t);
+
+      const topLeftSC_1 = traceLayer.getNewPoint!(topLeftSC_0.x, topLeftSC_0.y, 1);
+      const bottomRightSC_1 = traceLayer.getNewPoint!(bottomRightSC_0.x, bottomRightSC_0.y, 1);
+
+      const topLeftDC_t = { x: scales.x.invert(topLeftSC_1.x), y: scales.y.invert(topLeftSC_1.y) };
+      const bottomRightDC_t = { x: scales.x.invert(bottomRightSC_1.x), y: scales.y.invert(bottomRightSC_1.y) };
+
+      const tZoomExtentsDC: ZoomExtents = {
+        x: [topLeftDC_t.x, bottomRightDC_t.x],
+        y: [topLeftDC_t.y, bottomRightDC_t.y]
+      };
+
+      return this.closeSvgPath(tLineSegmentsSC, zoomExtentsSC, tZoomExtentsDC, tLineBoundaryInfo);
+    };
+  };
+
+  private closeSvgPath = (lineSegmentsSC: string[], zoomExtentsSC: ZoomExtents, zoomExtentsDC: ZoomExtents, lineBoundaryInfo: LineBoundaryInfo) => {
+    const { yCoordForXAxisSC, xMinSC, xMaxSC, xMinDC, xMaxDC, canvasPath } = lineBoundaryInfo;
+    const [ bottomSC, topSC ] = zoomExtentsSC.y;
+    const [ bottomDC, topDC ] = zoomExtentsDC.y;
+
+    // remember that in SC higher y values are lower on the screen
+    // this find the closest boundary to the x axis:
+    //   if x axis is in view, this is the x axis
+    //   if x axis is above the top of the svg, we treat the top extent as the x axis for closed area loops
+    //   if x axis is below the bottom of the svg, we treat the bottom extent as the x axis for closed area loops
+    let yCoordOfXAxisBoundarySC: number;
+    if (topSC < yCoordForXAxisSC && yCoordForXAxisSC < bottomSC) {
+      yCoordOfXAxisBoundarySC = yCoordForXAxisSC;
+    } else if (topSC >= yCoordForXAxisSC) {
+      yCoordOfXAxisBoundarySC = topSC;
+    } else {
+      yCoordOfXAxisBoundarySC = bottomSC;
+    }
+
+    const startXBoundarySC = Math.max(zoomExtentsSC.x[0], xMinSC);
+    const endXBoundarySC = Math.min(zoomExtentsSC.x[1], xMaxSC);
+
+    const startXBoundaryDC = Math.max(zoomExtentsDC.x[0], xMinDC);
+    const endXBoundaryDC = Math.min(zoomExtentsDC.x[1], xMaxDC);
+    const midXBoundaryDC = (startXBoundaryDC + endXBoundaryDC) / 2;
+
+    if (lineSegmentsSC.length > 0) {
+      const firstPointY = lineSegmentsSC[0].split("L")[0].substring(1).split(",").map(parseFloat)[1];
+      const firstPart = `M${startXBoundarySC},${yCoordOfXAxisBoundarySC}L${startXBoundarySC},${firstPointY}`;
```

Comment:
Adding one variable in, `xBoundaryStartSC`, to kind of make it legible where our path begins:

```suggestion
      const firstPointY = lineSegmentsSC[0].split("L")[0].substring(1).split(",").map(parseFloat)[1];
      const xBoundaryStartSC = { x: startXBoundarySC, y: yCoordOfXAxisBoundarySC };
      const firstPart = `M${xBoundaryStartSC.x},${xBoundaryStartSC.y}L${startXBoundarySC},${firstPointY}`;
```

By the way, is `L${startXBoundarySC},${firstPointY}` always going to be the same as grabbing the first point from the line segment? So can we grab that point whole instead of just its y coordinate?

Reply:
im not sure that variables makes it clearer, now there are two vars, `startXBoundarySC` and `xBoundaryStartSC` and i would have no idea what the difference is between these when i come back to this code

also `L${startXBoundary},${firstPointY}` is not guaranteed to be the first point in the line segment, for example a line segment that comes down from the top but away from the left boundary of the svg, then we would end up drawing a diagonal line between the x axis point and the first point in our line segment

Reply:
Fair